stl/inc/algorithm

```diff
@@ -606,26 +606,30 @@ namespace ranges {
         _NODISCARD constexpr iter_difference_t<_It> operator()(
             _It _First, _Se _Last, const _Ty& _Val, _Pj _Proj = {}) const {
             _Adl_verify_range(_First, _Last);
+            return _Count_unchecked(
+                _Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)), _Val, _Pass_fn(_Proj));
+        }
 
-            auto _UFirst                  = _Get_unwrapped(_STD move(_First));
-            const auto _ULast             = _Get_unwrapped(_STD move(_Last));
+        template <input_range _Rng, class _Ty, class _Pj = identity>
+            requires indirect_binary_predicate<ranges::equal_to, projected<iterator_t<_Rng>, _Pj>, const _Ty*>
+        _NODISCARD constexpr range_difference_t<_Rng> operator()(_Rng&& _Range, const _Ty& _Val, _Pj _Proj = {}) const {
+            return _Count_unchecked(_Ubegin(_Range), _Uend(_Range), _Val, _Pass_fn(_Proj));
+        }
+        // clang-format on
+    private:
+        template <class _It, class _Se, class _Ty, class _Pj>
+        _NODISCARD constexpr iter_difference_t<_It> _Count_unchecked(
+            _It _First, const _Se _Last, const _Ty& _Val, _Pj _Proj) const {

```

Comment:
Can we `_STL_INTERNAL_STATIC_ASSERT` the requirements here? (I know it seems silly, but I want to ensure these internal functions are in some way guarded against being used incorrectly under future maintenance. It's an attempt to assuage my guilt over stripping away perfectly good constraints. ;))

Reply:
Done